cpp/ps0004/p0004.cpp

```diff
@@ -1,33 +1,33 @@
 #include <iostream>
 using namespace std;
 int main() {
-    int number;
-   cout << "Enter a number between 32 and 126: ";
-      cin >> number;
+    // prompting user for number 1/1.5
+    float number;
+    cout << "Enter a number between 32 and 126: ";
+    cin >> number;
+
+    // checking invalidaty of user's input 4/4
     if (!(number)) {
         std::cout << "Invalid character/s or zero value." ;
         return 1;
-        // هنا جمله if 
-        //هتحدد اللي انا مدخله صح ولا غلط انا دلوقتي عملت الرقم عدد صحيح باستخدام int 
-       // اي رقم هدخله هيخلي الناتج ترو فبالتالي هيطبع ويخرج من البرنامج
-    //   انما لو دخلت صفر او اي حاجه غير الارقام هيبقي الناتج فولس وهتتنفي يبقي ترو ويطبع ويخرج خالص
+        // هنا جمله if
+        // هتحدد اللي انا مدخله صح ولا غلط انا دلوقتي عملت الرقم عدد صحيح باستخدام float
+        // اي رقم هدخله هيخلي الناتج ترو فبالتالي هيطبع ويخرج من البرنامج
+        // انما لو دخلت صفر او اي حاجه غير الارقام هيبقي الناتج فولس وهتتنفي يبقي ترو ويطبع ويخرج خالص
     }
 
-   if (number != int(number)) {
-       cout << "Invalid: a floating point number."  ;
-    }
-    
-    else if (number <= 0) {
-       cout << "Invalid character/s or zero value.";
+    if (number != int(number)) {
+       cout << "Invalid: a floating point number." ;
     }
     
     else if (number < 32 || number > 126) {
-        std::cout << "Invalid: number is out of the range." ;
+        cout << "Invalid: number is out of the range.";
     }

```

Comment:
- [x] used `if` structure in a right way
- [x] `if` structure take in consideration all invalid cases
- [x] all expressions to check invalidity are correct
- [x] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases